---
title: 'RNA_Seq_Methods_UpdatedMethods'
author: "Jose_Robledo"
date: "2024"
output:
  html_document:
    theme: darkly
    toc: true
    toc_float: true
    highlight: zenburn
---

# RNA-Sequencing experimental analysis workflow using *Caenorhabditis elegans*

**<font size="4">Jose Robledo, Saifun Ripa Nahar, Manuel A. Ruiz, Raymond J. Hendricks, David J. Burks, Mary L. Ladage, Taegun Kwon, Rajeev K. Azad, Pamela A. Padilla^1^</font>**

^1^Department of Biological Sciences, University of North Texas, Denton, TX 76203

Key Words: RNA sequencing, Bioinformatic pipeline, Differentially expressed genes, Reproducibility

<font size="4">Corresponding Authors:</font>
```{r eval=FALSE}
Corresponding Author:
Pamela A. Padilla (PI)
University of North Texas
Department of Biological Sciences
1155 Union Circle #305220
Denton, TX 76203
(940)565-3614
pamela.padilla@unt.edu

Jose Robledo (Bioinformatic Graduate Student)
University of North Texas
Department of Biological Sciences
1155 Union Circle #305220
Denton, TX 76203
JoseRobledo@my.unt.edu
```

# Abstract

<font size="4">RNA sequencing is an approach to transcriptomic profiling that enables the detection of differentially expressed genes in response to genetic mutation or experimental treatment, among other uses. Here we describe a method for the use of a customizable, user-friendly bioinformatics pipeline to identify differentially expressed genes in RNA sequencing data obtained from *C. elegans*, with attention to the improvement in reproducibility and accuracy of results. </font>

# 3. Methods

![](MiMB_fig1.png)
<font size="4">**Fig. 1 Experimental Workflow** </font>

## 3.4	Bioinformatic Pipeline: Linux Command Line (Terminal)

### Part I: Linux Command Line (Terminal)

<font size="4">

**Part 1** of the Bioinformatic pipeline is written for the use of a MacBook Pro using the Linux bash command line (e.g Terminal application). Create a folder where all bioinformatic analysis files are located for part 1 of the analysis (*see* **Note 22-27**). 

</font>

#### 3.4.1 Installing Required Programs

<font size="4">

1.	Use the Anaconda (Conda) repository environment management system to install all free open-source packages for Windows, macOS or Linux. Anaconda (Conda) installation can be downloaded free using the Google Chrome Internet Browser at [https://www.anaconda.com/]. Follow the directions to install the Conda environment on Linux Terminal provided by the User Guide at [https://docs.conda.io/projects/conda/en/latest/user-guide/index.html] (*see* **Note 28**).

2.	Search for the desired program on the Conda page at [https://www.anaconda.com/] or the Anaconda Navigator application. Copy and paste a Conda installer into the Linux terminal to run installation (*see* **Note 29**). 

3.	Install the following packages using a Conda environment:

- a.	FASTQC - https://anaconda.org/bioconda/fastqc 
- b.	FASTP - https://anaconda.org/bioconda/fastp [**5**]
- c.	SALMON - https://anaconda.org/bioconda/salmon 
- d.	STAR - https://anaconda.org/bioconda/star [**6**]
- e.	SAMTOOLS - https://anaconda.org/bioconda/samtools [**7,8**]
- **f. MultiQC - https://github.com/MultiQC/MultiQC ** (*see* **Updated Note 1**)

</font>

#### 3.4.2	FASTQ files 

<font size="4">

1.	The RNA Illumina Sequencing system for paired-end reads exports a total of 8 FASTQ files per sample. Plus and minus strands (R1 and R2) were sequenced for each lane (resulting in 4 lanes total) for each sample provided (*see* **Note 30**). To consolidate the FASTQ files for each sample, concatenate (Cat command in Linux) all lanes (001-004) for each strand (R1 and R2). This will result in 2 FASTQ files (R1 and R2) for each sample (*see* **Note 31**). 

</font>

**Updated Code**
```{bash eval=FALSE}

# sample 1 file
Sample Linux Code: 
~% cat /file/location/sample1_L001_R1.fastq /file/location/sample1_L002_R1.fastq /file/location/sample1_L003_R1.fastq /file/location/sample1_L004_R1.fastq > /file/location/sample1_R1.fastq 

Sample Linux Code: 
~% cat /file/location/sample1_L001_R2.fastq /file/location/sample1_L002_R2.fastq /file/location/sample1_L003_R2.fastq /file/location/sample1_L004_R2.fastq > /file/location/sample2_R2.fastq 
```


#### 3.4.3	Quality Control

<font size="4">

1.	Use the FASTQC package to perform quality control analysis by following the documentation directions at [https://www.bioinformatics.babraham.ac.uk/projects/fastqc/]. Quality control analysis of each sample FASTQ file was completed. Example reports to aid quality report interpretation are provided in the Example Reports section of the FASTQC documentation (*see* **Note 32**). **To generate a quality control report of all files run multiqc package by following the documentation directions [https://github.com/MultiQC/MultiQC]** (*see* **Updated Note 2**)

</font>
**Updated Code**
```{bash eval=FALSE}

# sample 1 file - prefiltering Quality Control
Sample Linux Code: 
~% FastQC /file/location/fastq_folder/*.fastq -o /file/location/qc_1 -t 4

############# additional code
multiqc .
```

#### 3.4.4	Adapter Trimming

<font size="4">

1.	Perform additional quality control and adapter trimming using the FASTP package following the documentation directions at [https://github.com/OpenGene/fastp#fastp] (*see* **Note 33**). Reports are generated to assess the quality of the data and can be interpreted using the documentation at [https://github.com/OpenGene/fastp#fastp] or by using FASTQC report examples. **To generate a quality control report of all files run multiqc package by following the documentation directions [https://github.com/MultiQC/MultiQC]** (*see* **Updated Note 2**)

</font>
**Updated Code**
```{bash eval=FALSE}

# sample 1 file
Sample Linux Code: 
~% fastp -w 4 -i /file/location/sample1_R1.fastq -o /file/location/fastp/sample1_R1.fastq -I /file/location/sample1_R2.fastq -O /file/location/fastp/sample1_R2.fastq -h /file/location//fastp/sample1_fastq.html -j /file/location//fastp/sample1_fastq.json

############# additional code
multiqc .
```

<font size="4">

2.	Quality control analysis is performed on both pre- and post- adapter trimmed FASTQ files to check for adapter contamination. FASTP performs the adapter trimming step and generates new FASTQ files that are used for the final analysis.

</font>

#### 3.4.5	Alignment / Mapping

<font size="4">

**1.**	Using Google Chrome, download the *Caenorhabditis elegans* genome assembly FASTA file (**GCF_000002985.6_WBcel235_genomic.fna** file extension) from the RefSeq NCBI website [https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000002985.6/] (*see* **Note 34 & Updated Note 3**).

**2.**	Using Google Chrome, download the *Caenorhabditis elegans* genome annotation file (**genomic.gtf** file extension) from the RefSeq NCBI website [https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000002985.6/] (*see* **Note 34 & Updated Note 3**).

**3.**	Create a “Bioinformatic analysis” folder for the experimental dataset being analyzed. Within, create a folder named “STAR_RefSeq_index.” Transfer the FASTA and annotation files from RefSeq NCBI into the “STAR_RefSeq_index” folder. 

4.	Generate a genome index for *C. elegans* using the STAR genomeGenerate command and follow the package documentation directions at [https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf]. For the small genome of *C. elegans*, set the –genomeSAindexNbases to 12. To reduce RAM consumption, set the –genomeChrBinNbits to 12 (*see* **Note 35**). 

</font>
**Updated Code**
```{bash eval=FALSE}

# STAR GenomeGenerate - Indexing C. elegans Genome
Sample Linux Code: 
~% STAR --runMode genomeGenerate --runThreadN 4 --genomeDir /folder/location/STAR_RefSeq_index --genomeFastaFiles /file/location/STAR_RefSeq_index/GCF_000002985.6_WBcel235_genomic.fna --genomeSAindexNbases 12 --genomeChrBinNbits 12 --outFileNamePrefix /folder/location/STAR_RefSeq_index/STAR_RefSeq_index --sjdbGTFfile /file/location/STAR_RefSeq_index/genomic.gtf --sjdbOverhang 100 

```

<font size="4">

**5.**	For each sample, run a STAR alignment using the default settings and follow the package documentation directions at [https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf]. Additional commands to include are the export of unmapped reads files, GeneCounts option (ReadsPerGene.out.tab files), BAM coordinated/uncoordinated files and the adjustments described in **step 4** (*see* **Note 34-37** & **Updated Note 4**).

</font>
**Updated Code**
```{bash eval=FALSE}

# STAR Alignment Code
Sample Linux Code: 
~% STAR --runThreadN 4 --genomeDir /folder/location/STAR_RefSeq_index --readFilesIn /file/location/fastp/sample1_trial1_R1.fastq /file/location/fastp/sample1_trial1_R2.fastq --quantMode TranscriptomeSAM GeneCounts --outSAMtype BAM Unsorted SortedByCoordinate --outFileNamePrefix /file/location/star_rawcounts/sample1_trial1 

```

<font size="4">

6.	Perform alignment on both pre- and post- trimmed data to examine adapter contamination. However, only use post-trimmed data for final analysis (*see* **Note 34-37**). 

</font>

#### 3.4.6	Validating Library Strandedness (IGV, SALMON and STAR genecount) (*see* **Note 38**)

<font size="4">

1.	To differentiate between a stranded and unstranded sample library, first install the Integrative Genomics Viewer (IGV) following the package documentation directions at [https://software.broadinstitute.org/software/igv/].

**2.**	Index all the STAR exported BAM coordinated files using SAMTOOLS following package documentation directions for indexing BAM files at [http://www.htslib.org/doc/samtools.html] (*see* **Updated Note 5**. 

</font>
**Updated Code**
```{bash eval=FALSE}

# Samtools index .bam coordinate files
Sample Linux Code: 
~% samtools index /file/location/sample1_trial1_Aligned.sortedByCoord.out.bam /file/location/sample1_trial1_Aligned.sortedByCoord.out.bam.bai

############# additional code
igvtools index ~/file/location/Sample1Aligned.sortedByCoord.out.bam ~/file/location/Sample1Aligned.sortedByCoord.out.bam.bai
```

<font size="4">

**3.**	Load the indexed BAM coordinated files onto IGV, select the *C. elegans* **ce11** genome and determine the strandedness of the library by following the instructions at [https://darencard.net/blog/2019-09-13-determining-stranded-rnaseq/ ] (*see* **Note 39**). The number of counts associated with each gene can also be explored using the genome browser.

**4.**	Next, to ensure that the library is properly analyzed, SALMON is used to detect the strand specificity of sequenced reads. Download the *C. elegans* genome FASTA file (**GCF_000002985.6_WBcel235_genomic.fna** file extension) again using the Google **Chrome RefSeq NCBI website files in [https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000002985.6/]** and place into a folder called “wormgenome_salmon”(*see* **Note 34 & Updated Note 3**).

5.	Index the *C. elegans* genome FASTA file following the SALMON package documentation directions at [https://salmon.readthedocs.io/en/latest/] (*see* **Note 40**).

</font>
**Updated Code**
```{bash eval=FALSE}

# SALMON Index C.elegans genome
Sample Linux Code: 
~% salmon index -t /file/location/salmon/wormgenome_salmon/GCF_000002985.6_WBcel235_genomic.fna -i /file/location/salmon/wormgenome_salmon

```

<font size="4">

6.	Using one of the previously generated, post-trim sample FASTQ files, quantify gene counts in mapping-based mode following the SALMON package documentation directions at [https://salmon.readthedocs.io/en/latest/]. Determine the strandedness of the library by interpreting the acronym output of the library using the package documentation (*see* **Note 41**). 

</font>
**Updated Code**
```{bash eval=FALSE}

# SALMON run pseudoaligner 
Sample Linux Code: 
~% salmon quant -i /file/location/salmon/wormgenome_salmon -l A -1 /file/location/fastp/sample1_trial1_R1.fastq -2 /file/location/fastp/WT_trial1_R2.fastq --gcBias --validateMappings -o transcript_quant

# Output
Automatically detected most likely library type as "see salmon documentation for library codes"

############# additional code
# SALMON run pseudoaligner 
Sample Linux Code: 
~% salmon quant -i /file/location/salmon/wormgenome_salmon -l A -r /file/location/salmon/fastp/sample1_trial1.fastq --validateMappings -o transcript_quant_sample1_trial1

# Output
Automatically detected most likely library type as U (Unstranded Data)
```

<font size="4">

7.	To further validate the strandedness of the samples the STAR genecount option from **step 5** exports a text file (extension “ReadsPerGene.out.tab”) with four columns corresponding to strandedness possibilities: gene ID, counts for unstranded RNA-seq, counts for first read strand aligned with RNA, and counts for second read strand aligned with RNA. Use a grep/awk script command in the Linux terminal to determine the number of reads in each column for each sample in ReadsPerGene.out.tab text file. The count method here can be referenced in [https://biocorecrg.github.io/RNAseq_course_2019/alnpractical.html] and strandedness can be determined by the column with the most reads (*see* **Note 42** and **43**). 

</font>

```{bash eval=FALSE}

# Count the number of reads from STAR output files
Sample Linux Code: 
~% grep -v "N_" /file/location/star_rawcounts/Sample1_trial1ReadsPerGene.out.tab | awk '{unst+=$2;forw+=$3;rev+=$4}END{print unst,forw,rev}'

```

#### 3.4.7	STAR genecount processed files

<font size="4">

1.	Once library strandedness is determined, follow the STAR package documentation to extract the appropriate columns from the ReadsPerGene.out.tab text files for all samples exported in **step 5**. Column 1 reports the gene list and columns 2-4 report gene counts corresponding to library strandedness. Perform the processing of ReadsPerGene.out.tab text files for all samples using a cut command to extract column 1 and the appropriate second column onto a new text file; this file will be used for all further analysis (*see* **Note 43-44**). 

</font>
**Updated Code**
```{bash eval=FALSE}

# process STAR genecount txt files
Sample Linux Code: 
~% cut -f1,4 /file/location/star_rawcounts/Sample1_trial1ReadsPerGene.out.tab > Sample1_trial1_counts.txt

```

## 3.5	 Bioinformatic Pipeline - R and R-studio

### Part II: R and R-studio

<font size="4">

**Part 2** of the Bioinformatic pipeline utilizes R and R-studio on a MacOS/Linux platform. Multiple free R packages are available to analyze RNA-sequencing data. The R language can perform statistical analysis and graphing analysis (*see* **Note 22**). 

1.	Create a folder where all R/R-studio analysis will be performed for the project. 

2.	Inside the R project folder, create a folder labeled “star_rawcounts” and transfer all the processed STAR files from Subheading **3.4.7** (*see* **Note 44**).

3.	Create a “sample_description.txt” file with a description of the samples, genotypes, sample names, file names (e.g. from STAR raw counts file folder), trial numbers, etc. (*see* **Note 45**).

**4.**	Using a previously downloaded *Caenorhabditis elegans* genome annotation file (**GCF_000002985.6_WBcel235_genomic.fna** file extension) from the RefSeq NCBI website [https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000002985.6/] (*see* **Note 34 & Updated Note 3**) create a “wormgenes.txt” file that includes all *C. elegans* genes with associated RefSeqIDs. Extracting wormgenes and WormbaseIDs using the *C. elegans* annotation file from RefSeq can be done by using a cat and awk command in the Linux Terminal application (*see* **Note 46**). Alternatively, can use the geneInfo.tab from *C. elegans* genome index generated by STAR Subheading **3.4.5 step 4** (*see* **Updated note 3 & 6**).

</font>
**Updated Code**
```{bash eval=FALSE}

# process wormgenes.txt file from STAR Genome Index Files
Sample Linux Code: 
~% { grep "CELE_" geneInfo.tab ; grep "KEF34_" geneInfo.tab; } > wormgenes.refseq.294.txt

```

<font size="4">

**5.**	Download a “functional_description.txt” file from the Wormbase website [https://downloads.wormbase.org/releases/current-production-release/] with file name “c_elegans.PRJNA13758.WS294.functional_descriptions.txt.gz”. 

6.	The functional_description.txt file will be used to extract the following descriptions: WBgenes, gene class description, concise description, and automated description. Use multiple awk commands to create 4 functional gene description text files using WBgenes and a different Wormbase gene description column (i.e. class description, concise description, or automated description) (*see* **Note 47**).

</font>
**Updated Code**
```{bash eval=FALSE}

# process functional descriptions txt file
Sample Linux Code:
gene list: 
~% grep "WBGene" c_elegans.PRJNA13758.WS294.functional_descriptions.txt > WB294_WBgenes1.txt 
gene class description: 
~% awk '/Gene class description:/' c_elegans.PRJNA13758.WS294.functional_descriptions.txt > WB294_geneclass.txt
concise description: 
~% awk ' /Concise description:/ { flag=1; pfx="" } /Automated description/ { flag=0; print "" } flag { printf "%s%s",pfx,$0; pfx=" " } ‘ c_elegans.PRJNA13758.WS294.functional_descriptions.txt > WB294_concise.txt

automated description:
~% awk ' /Automated description:/ { flag=1; pfx="" } /Gene class description:/ { flag=0; print "" } flag { printf "%s%s",pfx,$0; pfx=" " } ' c_elegans.PRJNA13758.WS294.functional_descriptions.txt > WB294_automated.txt

```

<font size="4">

7.	Download and install R using the website directions at [https://cran.r-project.org/]. Download and Install R-studio Desktop using the website directions at [https://posit.co/download/rstudio-desktop/]. Open R-studio Desktop, create a new project and assign the folder created in **step 1** as the directory. All further analysis to identify DEGs use R-studio Desktop or R. 

8.	Install all the following library packages required to identify Differentially Expressed Genes (DEGs) by following package documentation directions (*see* **Note 48**):

- a.	Devtools - https://www.r-project.org/nosvn/pandoc/devtools.html 
- b.	DESeq2 - https://bioconductor.org/packages/release/bioc/html/DESeq2.html 
- c.	Ashr - https://github.com/stephens999/ashr 
- d.	Dplyr - https://cran.r-project.org/web/packages/dplyr/dplyr.pdf 
- e.	Writexl - https://cran.r-project.org/web/packages/writexl/index.html 
- f.	Ggplot2 - https://ggplot2.tidyverse.org/index.html 
- g.	Ggrepel - https://cran.r-project.org/web/packages/ggrepel/index.html 
- h.	Pheatmap - https://www.r-bloggers.com/2022/10/the-pheatmap-function-in-r/ 
- i.	PcaExplorer - https://bioconductor.org/packages/release/bioc/html/pcaExplorer.html 

9.	Load all library packages required for the desired analysis to identify DEGs. 

</font>

```{r eval=FALSE}

# load package libraries
Sample R Code: 
> library(devtools) #install.packages(“devtools”)
> library(DESeq2) #BiocManager::install(“DESeq2”)
> library(ashr) #install_github(“stephens999/ashr”)
> library(dplyr) #install.packages(“dplyr”)
> library(writexl) #install.packages(“writexl”)
> library(ggplot2) #install.packages(“ggplot2”)
> library(tidyverse) #install.packages(“tidyverse”)
> library(ggrepel) #install.packages(“ggrepel”)
> library(pheatmap) #install.packages(“pheatmap”)
> library(pcaExplorer) #BiocManager::install(“pcaExplorer”)

```

<font size="4">

10.	Import the sample_description.txt file using the read.table function in R-studio (*see* **Note 49**). 

</font>

```{r eval=FALSE}

# import sample_description txt file
Sample R Code: 
> sample_description <- read.table("sample_description.txt", header=T, sep="\t")

```

<font size="4">

11.	Import the wormgenes.txt file using the read.table function in R-studio (*see* **Note 49**). 

</font>

```{r eval=FALSE}

# import wormgenes txt file
Sample R Code: 
> wormgenes <- read.table("wormgenes.ensembl.106.txt", header=FALSE, sep="\t")

```

<font size="4">

12.	Import Wormbase functional gene description text files from **step 6** using read.table function in R-studio (*see* **Note 49**).

</font>
**Updated Code**
```{r eval=FALSE}

# import WormBase functional description files
Sample R Code: 
> worm1 <- read.table("WB294_WBgenes1.txt", header=F, sep="\t", rownames(F))
> worm2 <- read.table("WB294_geneclass.txt", header=F, sep="\t", rownames('Gene class description:'))
> worm3 <- read.table("WB294_concise.txt", header=F, sep="\n", rownames('Concise description'))
> worm4 <- read.table("WB294_automated.txt", header=F, sep="\t", rownames('Automated description:'))

```

<font size="4">

13.	Combine all Wormbase functional gene descriptions onto a new dataframe using the cbind function (4 columns total). Delete the extra column names found on each gene description so that only the gene description text for each column remains (*see* **Note 50-51**). 

</font>
**Updated Code**
```{r eval=FALSE}

# combine functional description files - further edit for merging later - remove extra characters
Sample R Code: 
> wormgene_desc <- cbind(worm1,worm2,worm3,worm4)
> colnames(wormgene_desc)
> colnames(wormgene_desc) <- c("WormBaseID", "GeneName", "Transcript_Name", "Gene_class_description", "Concise_description", "Automated_description")
> wormgene_desc$Gene_class_description <- gsub("Gene class description: ","",wormgene_desc$Gene_class_description)
> wormgene_desc$Concise_description <- gsub("Concise description: ","",wormgene_desc$Concise_description)
> wormgene_desc$Automated_description <- gsub("Automated description: ","",wormgene_desc$Automated_description)

```

<font size="4">

14.	Export this data as an Excel file using the write_xlsx function. This is the final Wormbase functional gene description file (*see* **Note 52**). 

</font>
**Updated Code**
```{r eval=FALSE}

# export dataframe as excel file
Sample R Code: 
> write_xlsx(wormgene_desc, "wormbase.WS294_genedescriptions.xlsx")

```

<font size="4">

**14.1** Combine RefSeq data with WormBase data. First will need to edit columns so that both sheets match without losing the origininal text (*see* **Updated Note 7**). 

</font>
**Updated Code**
```{r eval=FALSE}
############# Parse Wormbase.org gene file in R

# edit WormBase.org transcriptIDs for merge two files by using matching transcript names

# Wormbase.org and RefSeq have different number of genes
> nrow(worm1)
# WormBase functional descriptions files contain 52,109 genes
> nrow(wormgenes) 
# RefSeq geneome assembly contain 46,928 genes
# Wormbase - RefSeq genes: 52,109 - 46,928 = 5,181 different genes

############# 1. Make a new column name on both RefSeq & WormBase #############
# edit wormbase worm1 column names & RefSeq wormgenes column names
> colnames(worm1) <- c("WormBaseID", "GeneName", "WormBase_Transcript")

> wormgenes <- wormgenes[, -3] # remove unnecessary column 3
> wormgenes$V2 <- gsub("CELE_","",wormgenes$V2) #delete extra characters in each row - this leaves transcript name
> colnames(wormgenes) <- c("RefSeq_genename", "RefSeq_Transcript") # edit column names

# create a merged transcript column to combine WormBase & RefSeq
> worm1$merge_transcript = worm1$WormBase_Transcript
> wormgenes$merge_transcript = wormgenes$RefSeq_Transcript


############# 3. Merge RefSeq & WormBase files #############
> refseqWB294_genes <- merge(worm1, wormgenes, by = "merge_transcript", all = F)
> nrow(wormgenes) # only 41,200 genes were merged from 2 databases - 46,928 aligned genes is our goal to match


############# 4. Determine which genes were not merged & edit merge_transcript columns #############
> missing_genes <- anti_join(worm1, wormgenes, by = "merge_transcript") # anti_join will do the opposite of merge and provide you with a dataframe including all the genes that did not merge
> nrow(missing_genes) # 10,909 genes not merging
# 4,094 Wormbase genes have "NOT KNOWN" transcript IDs - we are not interested in these genes
> sum(worm1$WormBase_Transcript == 'not known') # 4,094 WormBase_TranscriptIDs are not known

# the rest of the genes have differing transcript names - isoforms for example worm1 = T13A10.10a & wormgenes = T13A10.10
# remove extra a isoform character on each transcript in worm1 file
> worm1$merge_transcript <- gsub("a","",worm1$merge_transcript)
> worm1$merge_transcript <- gsub("b","",worm1$merge_transcript)
> worm1$merge_transcript <- gsub("c","",worm1$merge_transcript)
> worm1$merge_transcript <- gsub("d","",worm1$merge_transcript)
# just one transcript has isoform e - manually be done

# to ensure proper merging - capitalize merge_transcript columns on both files
> worm1$merge_transcript <- toupper(worm1$merge_transcript) # upper-case conversion to ensure combining files
> wormgenes$merge_transcript <- toupper(wormgenes$merge_transcript) # upper-case conversion to ensure combining files

# merge files again
> refseqWB294_genes <- merge(worm1, wormgenes, by = "merge_transcript", all = F)
> nrow(refseqWB294_genes) # 46867 genes successfully merged and now we have 61 missing genes 

# identify missing genes that refseq is not merging with wormbase
> missing_genes <- anti_join(wormgenes, worm1, by = "merge_transcript") # anti_join will do the opposite of merge and provide you with a dataframe including all the genes that did not merge
> nrow(missing_genes) # 61 genes 

> View(missing_genes) # view file and search for each of these genes in worm1 and wormgenes variables

# fixing 61 gene annotations from refseq to match with wormbase genes - wormgenes edit file
# 15 TEL54X.2 genes are annotated in refseq with cTEL54X.2 - (61 - 15 = 46)
# 4 genes have isoform issues - F59E12.6_1 F59E12.6_3 F59E12.6_4 K09C6.2 (46 - 4 = 42)
# 6 genes have gene names instead of transcript IDs (42 - 6 = 36)
# 36 genes are transposons, mitochondria genes - KEF34_t01 or KEF34_p12 - (36 - 36 = 0): I will analyze these list separately

############# 4. Edit merge_transcript columns #############


# edit 15 TEL54X.2 gene transcriptIDs in each file
# WormBase genes
> letter_C <- c("TEL55X.1","TEL7X.1","TEL54X.1","TEL17.1","TEL7X.2","TEL3X.2","TEL3X.3","TEL29B.1","TEL17.2","TEL79B.1","TEL52S.1","TEL7X.3","TEL29B.2","TEL79B.2","TEL54X.2")
> letter_C_replace <- c("cTel55X.1","cTel7X.1","cTel54X.1","cTel17.1","cTel7X.2","cTel3X.2","cTel3X.3","cTel29B.1","cTel17.2","cTel79B.1","cTel52S.1","cTel7X.3","cTel29B.2","cTel79B.2","cTel54X.2")
> worm1$merge_transcript <- replace(worm1$merge_transcript, worm1$merge_transcript %in% letter_C, letter_C_replace)

# RefSeq genes
> letter_C2 <- c("CTEL29B.1", "CTEL29B.2", "CTEL52S.1", "CTEL54X.1", "CTEL54X.2", "CTEL79B.1", "CTEL79B.2", "CTEL3X.3", "CTEL3X.2","CTEL17.2", "CTEL17.1", "CTEL7X.3", "CTEL7X.2", "CTEL7X.1", "CTEL55X.1")
> letter_C2_replace <- c("cTel29B.1", "cTel29B.2", "cTel52S.1", "cTel54X.1", "cTel54X.2", "cTel79B.1", "cTel79B.2", "cTel3X.3", "cTel3X.2","cTel17.2", "cTel17.1", "cTel7X.3", "cTel7X.2", "cTel7X.1", "cTel55X.1")
> wormgenes$merge_transcript <- replace(wormgenes$merge_transcript, wormgenes$merge_transcript %in% letter_C2, letter_C2_replace)


# edit 4 isoform issues
> isoforms1 <- c("K09C6.2E")
> isoforms1_replace <- c("K09C6.2")
> worm1$merge_transcript <- replace(worm1$merge_transcript, worm1$merge_transcript %in% isoforms1, isoforms1_replace)

# isoforms 2 - 3 replicates that correspond to the same WormBaseID
> isoforms2 <- c("F59E12.6_1", "F59E12.6_3", "F59E12.6_4")
> isoforms2_replace <- c("F59E12.6", "F59E12.6", "F59E12.6")
> wormgenes$merge_transcript <- replace(wormgenes$merge_transcript, wormgenes$merge_transcript %in% isoforms2, isoforms2_replace)


# several gene names instead of gene transcripts - 6 genes
> gene_names <- c("LEV-10", "EAT-18", "CHA-1", "UNC-17", "EXOS-4.1", "TIN-9.2")
> gene_names_replace <- c("lev-10", "eat-18", "cha-1", "unc-17", "exos-4.1", "tin-9.2")
> wormgenes$merge_transcript <- replace(wormgenes$merge_transcript, wormgenes$merge_transcript %in% gene_names, gene_names_replace)

# change 6 genes manually in worm1 file
> worm1$merge_transcript <- replace(worm1$merge_transcript, 2912, "lev-10")
> worm1$merge_transcript <- replace(worm1$merge_transcript, 1082, "eat-18")
> worm1$merge_transcript <- replace(worm1$merge_transcript, 6683, "unc-17")
> worm1$merge_transcript <- replace(worm1$merge_transcript, 421, "cha-1")
> worm1$merge_transcript <- replace(worm1$merge_transcript, 7128, "exos-4.1")
> worm1$merge_transcript <- replace(worm1$merge_transcript, 24482, "tin-9.2")

############# 5. Merge merge_transcript columns #############

# final merge
> refseqWB294_genes <- merge(worm1, wormgenes, by = "merge_transcript", all = F)
> nrow(refseqWB294_genes) # 46,892 and only 36 genes are missing

# mitochondrial genome - KEF34 genes
> refseq294_mitogenome <- anti_join(wormgenes, refseqWB294_genes, by = "merge_transcript")


############# 6. Save an excel sheet of all aligned genes #############

# create xlsx files
> write_xlsx(refseqWB294_genes, "refseqWB294_genes.xlsx")
> write_xlsx(refseq294_mitogenome, "refseq294_mitogenome.xlsx")


############# 7. Create an all_worm_genes which include all transcript annotations #############

# merge final annotation file
> all_worm_genes <- merge(refseqWB294_genes, wormgene_desc, by = "WormBaseID", all = F)

# edit all_worm_genes column names
> colnames(all_worm_genes)
# [1] "WormBaseID"             "merge_transcript"       "GeneName.x"             "WormBase_Transcript"    "RefSeq_genename"        "RefSeq_Transcript"      "GeneName.y"            
# [8] "Transcript_Name"        "Gene_class_description" "Concise_description"    "Automated_description"

# remove unnecessary columns
> all_worm_genes <- all_worm_genes[,c(-2,-4,-7,-8)]
> colnames(all_worm_genes)
# [1] "WormBaseID"             "GeneName.x"             "RefSeq_genename"        "RefSeq_Transcript"      "Gene_class_description" "Concise_description"    "Automated_description"


# edit column names
> colnames(all_worm_genes) <- c("WormBaseID", "Gene_Symbol", "RefSeq_ID", "Transcript", "Gene_class_description", "Concise_description", "Automated_description")
# reorder columns
> all_worm_genes <- all_worm_genes[c("RefSeq_ID","WormBaseID","Transcript","Gene_Symbol","Gene_class_description","Concise_description","Automated_description")]
# convert to data frame & export file as excel sheet
> all_worm_genes <- as.data.frame(all_worm_genes)
> write_xlsx(all_worm_genes, "allwormgenes_refseqWB294.xlsx")

```

<font size="4">

15.	Construct a DESeq2 Data Set Object using the DESeqDataSetFromHTSeqCount function. Load in the sample_description data frame, directory for processed STAR files (Subheading 3.5**step 2**), and design of the experiment using a column(s) in the sample_description data frame (*see* **Note 53-54** & **Updated Note 8**)). 

</font>

```{r eval=FALSE}

# contruct DESeq2 Data Set Object using GeneCount files
Sample R Code: 
> experiment1 <- DESeqDataSetFromHTSeqCount(sampleTable = sample_description, directory = "star_rawcounts", design = ~ Genotype)

```

<font size="4">

16.	Filter the resulting data by removing genes with less than 10 summed raw counts [**9**] (*see* **Note 55**). 

</font>

```{r eval=FALSE}

# Optional: remove lowly expressed genes - 10 counts or less across experiments
Sample R Code:
> experiment1 <- experiment1[rowSums(counts(experiment1)) > 10, ]

```

<font size="4">

17.	Fit statistical model of DESeq2 using the DESeq function (*see* **Note 56**). 

</font>

```{r eval=FALSE}

# fit DESeq2 statistical model
Sample R Code: 
> stat1 <- DESeq(experiment1)

```

<font size="4">

18.	Create a norm_counts data frame for normalized counts using the log2(counts) function of the statistical model variable. Add WormbaseIDs and gene names to the norm_counts data frame using the merge(unique) function (*see* **Note 57-58**).

</font>
**Updated Code**
```{r eval=FALSE}

# normalized counts for experiment
Sample R Code: 
> norm_counts <- log2(counts(stat1, normalized = TRUE)+1) 
> norm_counts_genes <- merge(unique(all_worm_genes[,1:2]), data.frame(ID=rownames(norm_counts), norm_counts), by=1, all=F)

```

<font size="4">

19.	Using the norm_counts data frame, create a new data frame(s) for each sample to include the following columns: WormBaseIDs, gene name, sample 1 normalized counts for each trial replicate using base R functions (*see* **Note 59-60**). Use the rowMeans function to add a column of average normalized counts to each sample data frame (*see* **Note 61**). 

</font>
**Updated Code**
```{r eval=FALSE}

# Add geneIDs and Average counts per experiment
Sample R Code: 
sample1 <- norm_counts_genes[,c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)] #sample1
sample2 <- norm_counts_genes[,c(1, 2, 3, 4, 5, 6, 7, 11, 12, 13)] #sample2

sample1$sample1_avg_ncounts <- rowMeans(sample1[,8:10]) #sample1
sample2$glucose_avg_ncounts <- rowMeans(glucose[,8:10]) #sample2

sample1_avg_ncounts <- sample1[,c(1,2,4,5,6,7,11)] #sample1
sample2_avg_ncounts <- sample2[,c(1,2,4,5,6,7,11)] #sample2

avg_norm_counts <- merge(sample1_avg_ncounts, sample2_avg_ncounts)

```

<font size="4">

20.	Export the average normalized counts as a text file using the write.table function and as an Excel table using the write_xlsx function (*see* **Note 62**). 

</font>

```{r eval=FALSE}

# export average normalized counts
Sample R Code: 
> write.table(norm_counts_genes, "experiment1_normcounts.txt", quote=F, col.names=T, row.names=F, sep="\t")
> write_xlsx(norm_counts_genes, "experiment1_normcounts.xlsx")
> write.table(avg_norm_counts, "experiment1_avg_normcounts.txt", quote=F, col.names=T, row.names=F, sep="\t")  
> write_xlsx(avg_norm_counts, "experiment1_avg_normcounts.xlsx")

```

#### 3.5.1	Visualization – Principal Component Analysis and Sample Distance Heatmap

<font size="4">

1.	To visualize the variability between samples, perform a variance stabilizing transformation (vst function) using the statistical model of DESeq2. Calculate between sample distances matrix using the vst transformed variable and create a heatmap (*see* **Note 63-64**). 

</font>

```{r eval=FALSE}

# vst transform data and sample distance heatmap
Sample R Code: 
> vsd <- vst(stat1)
> sampleDistMatrix <- as.matrix(dist(t(assay(vsd))))
> png("sample_distance_heatmap.png")
> pheatmap(sampleDistMatrix)

```

<font size="4">

2.	Create a Principal Component Analysis (PCA) plot (plotPCA function) using the transformed variable and a sample column description from the sample_description data frame (Subheading **3.5**, **step 3**) (*see* **Note 65-67**). 

</font>

```{r eval=FALSE}

# vst transform data and plot PCA
Sample R Code: 
> png("PCA1.png")
> plotPCA(object = vsd, intgroup = "Genotype_ortreatment")

```

#### 3.5.2	Differentially Expressed Genes

<font size="4">

1.	To identify DEGs between 2 groups, use the results function and include the object statistical model of DESeq2, using a contrast argument to include the 2 comparison names (*see* **Note 68** and **69**). 

</font>

```{r eval=FALSE}

# DEG results for each group comparison
Sample R Code: 
> deg1 <- results(object = stat1, contrast=c("Genotype_ortreatment","sample1_treatment1","sample1_treatment2"))

```

<font size="4">

2.	Use the lfcShrink function and include the object statistical model of DESeq2, and a contrast argument to include the 2 comparison names (*see* **Note 68-69**). 

</font>
**did not perform in this analysis - only if wanting to perform figures using log2FC**
```{r eval=FALSE}

# shrink Log2FC - optional
Sample R Code: 
> deg1_shrink <- lfcShrink(dds = stat1, contrast=c("Genotype_ortreatment","sample1_treatment2","sample1_treatment1"), type="ashr")

```

<font size="4">

3.	Merge the wormgene description data frame (Subheading **3.5**, *step* **1-4**) with the lfcShrink variable using gene names as a common column in both data frames (*see* **Note 70**).

</font>
**Updated Code**
```{r eval=FALSE}

# add gene IDs to DEG file
Sample R Code: 
> deg1_genes <- merge(unique(wormgene_desc[1:7]), data.frame(ID=row.names(deg1), deg1), by=1, all=T)

```

<font size="4">

4.	Merge the average norm counts (Subheading **3.5**, *step* **19-20**) for the samples to the lfcShrink variable (*see* **Note 71**). Organize columns in the order desired (*see* **Note 72**).

</font>
**Updated Code**
```{r eval=FALSE}

# add average normalized counts for each sample comparison to DEG file and reorganize columns
Sample R Code: 
> deg1_genes <- merge(deg1_genes, avg_norm_counts)
> colnames(deg1_genes)  
> deg1_genes <- deg1_genes[c("WormBaseID", "RefSeq_ID","Transcript", "Gene_Symbol", "baseMean", "sample1_treatment1_avg_ncounts", "sample1_treatment2_avg_ncounts", "log2FoldChange", "lfcSE", "stat", "pvalue", "padj", "Gene_class_description", "Concise_description", "Automated_description")]

```

<font size="4">

5.	Export the DEGs of each comparison to a text file using the write.table function and as an Excel file using the write_xlsx function (*see* **Note 73**).

</font>


```{r eval=FALSE}

# export txt and excel file of results
Sample R Code: 
> write.table(deg1_genes, "deseq2_sample1treatment2_vs_sample1treatment1.txt", quote=F, col.names=T, row.names=F, sep="\t")
> write_xlsx(deg1_genes, "deseq2_sample1treatment2_vs_sample1treatment1.xlsx")

```

<font size="4">

Repeat **steps 1-5** from Subheading **3.5.2** for each comparison made for all samples. Each file should include the following columns: WormBaseIDs, Gene Name, Sample1 Average Norm Counts, Sample2 Average Norm Counts, log2FoldChange, lfcSE, stat, pvalue, padj, gene class description, gene concise description, and gene automated description.

</font>


# End